task/transcode.go

```diff
@@ -115,6 +116,24 @@ func fileWriter(size int64) WriteSeekCloser {
 	}
 }
 
+type SegmentCounter struct {
+	size  uint64
+	count uint64
+}
+
+func NewSegmentCounter(size int64) *SegmentCounter {
+	return &SegmentCounter{size: uint64(size)}
+}
+
+func (c SegmentCounter) Count() uint64 {
+	return atomic.LoadUint64(&c.count)
+}
+
+func (c SegmentCounter) Read(data []byte) int {
```

Comment:
So this function on the `ReadCounter` was only called this because it is made for wrapping an `io.Reader` into another `Reader`, and so it must follow that interface. Here, the `Read` name seems inaccurate and we don't need to return anything either, so we don't really need to make a similar interface.

So WDYT of actually making this an `Accumulator` instead which only receives an `uint64` and adds to its internal state var? We also:
 - Don't need any `size` variable here. It is not used in its internal logic and only constructed with it to be read once on the next line, so we don't really need it.
 - Can use the accumulator from `ReadCounter` and avoid the `atomic` logic duplication! You can even make it an anonymous field (go's most similar way of "inheriting") of `ReadConter`, so the `Count` function will already be exposed and you only need to call `Add` (or `Accumulate`, or sth like that) from its `Read` method.